src/components/client/data/RawData.tsx

```diff
@@ -14,33 +21,52 @@ export const RawData: React.FC = () => {
 
   const toast = useToast();
 
-  const onToggle = () => {
-    update((state) => {
-      if (encoding === "hex") {
+  const onChange = (encoding: string) => {
+    if (encoding === "hex") {
+      update((state) => {
         state.data.raw.content = bs58.encode(Buffer.from(content, "hex"));
-        state.data.raw.encoding = "bs58";
-      } else {
-        try {
-          state.data.raw.content = Buffer.from(
-            bs58.decode(state.data.raw.content)
-          ).toString("hex");
-          state.data.raw.encoding = "hex";
-        } catch (e) {
-          // it is not a valid base58 string
-          toast({
-            title: "Invalid string",
-            description: "This is not a valid base58 encoded string.",
-            status: "error",
-            isClosable: true,
-            duration: 30_000,
-          });
-        }
```

Comment:
encoding here is the radio button that the user is switcing to. so it should be if encoding is hex, we decode the bs58 and encode to hex.